### DISCUSSION TITLE
Facing connection refused error when trying to access python kubernetes client in elastalert

### DISCUSSION THREAD

**Discussion by Ramya1507** [jertel/elastalert2#621] [Q&A]
Hello,
I am new to the kubernetes python- client and I am trying to explore this elastalert  by using python-client  to get the configmap details so, I have added the **below changes** in the elastalert/elastalert.py,  I am running this in kubernetes cluster
`````
 def main(args=None):
      try:
           config.load_incluster_config()
      except:
           config.load_kube_config()
   
      v1 = client.CoreV1Api()
      namespace = "default"
      name = "example"
      try: 
          configmap = v1.read_namespaced_config_map(name,namespace)
          logging.info(configmap)
      except:
          logging.info("no configmap with the given details")
      signal.signal(signal.SIGINT, handle_signal)
      if not args:
        args = sys.argv[1:]
     client = ElastAlerter(args)
`````````````

I am facing the below warning even though configmap is present in the cluster, can anyone help me on this

**WARNING:urllib3.connectionpool:Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f675f0943c8>: Failed to establish a new connection: [Errno 111] Connection refused',)': /api/v1/namespaces/**

**Comment by BuffaloWill** [comment]
I am not familiar with the k8s python client, but are you trying to pull the config map from inside the pod via the k8s API?
